x11-wm/mutter/mutter-47.4-r1.ebuild

```
@@ -0,0 +1,253 @@
+	>=x11-libs/libX11-1.7.0
+	>=x11-libs/libXcomposite-0.4
+	x11-libs/libXcursor
+	x11-libs/libXdamage
+	x11-libs/libXext
+	>=x11-libs/libXfixes-6
+	>=x11-libs/libXi-1.7.4
+	x11-libs/libxkbfile
+	x11-misc/xkeyboard-config
+	x11-libs/libXrender
+	x11-libs/libxcb:=
+	x11-libs/libXinerama
+	x11-libs/libXau
+"
+
+RDEPEND+="
+	X? (
+	   ${X_OR_XWAYLAND_DEPS}
+	   x11-libs/libICE
+	   >=x11-libs/libXrandr-1.5.0
+	   >=x11-libs/libxkbcommon-0.4.3[X]
+	   x11-libs/libXtst
+	)
+	wayland? ( xwayland? ( ${X_OR_XWAYLAND_DEPS} ) )
+"
+
+DEPEND="${RDEPEND}
+	x11-base/xorg-proto
+	sysprof? ( >=dev-util/sysprof-common-3.38.0 )
+"
+BDEPEND="
+	dev-util/wayland-scanner
+	dev-util/gdbus-codegen
+	dev-util/glib-utils
+	>=sys-devel/gettext-0.19.8
+	virtual/pkgconfig
+	gtk-doc? ( >=dev-util/gi-docgen-2021.1 )
+	test? (
+		${PYTHON_DEPS}
+		$(python_gen_any_dep '
+			>=dev-python/python-dbusmock-0.28[${PYTHON_USEDEP}]
+		')
+		app-text/docbook-xml-dtd:4.5
+		X? ( x11-misc/xvfb-run )
+	)
+	wayland? (
+		>=sys-kernel/linux-headers-4.4
+		x11-libs/libxcvt
+	)
+"
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-43.0-Disable-anonymous-file-test.patch
+)
+
+python_check_deps() {
+	if use test; then
+		python_has_version ">=dev-python/python-dbusmock-0.28[${PYTHON_USEDEP}]"
+	fi
+}
+
+src_configure() {
+	use debug && EMESON_BUILDTYPE=debug
+	local emesonargs=(
+		# Mutter X11 renderer only supports gles2 and GLX, thus do NOT pass
+		#
+		#   -Dopengl_libname=libOpenGL.so.0
+		#
+		# while we build the x11 renderer, as we currently enable gles2 only
+		# with USE=wayland and x11 renderer wouldn't find the needed GLX symbols
+		# in a configuration where wayland is disabled, as libOpenGL doesn't
+		# include them.
+		#
+		# See
+		# - https://bugs.gentoo.org/835786
+		# - https://forums.gentoo.org/viewtopic-p-8695669.html
+
+		-Dopengl=true
+		$(meson_use wayland gles2)
+		#gles2_libname
+		-Degl=true
+		$(meson_use X glx)
+		$(meson_use wayland)
+	)
+
+	if use wayland; then
```

You added this because you prefer to silently ignore "xwayland" flag when wayland is not enabled over adding "REQUIRED_USE=xwayland? ( wayland )"? 🤔

---

The weird syntax in this case is a consequence of the package.use.force. When the wayland flag is disabled, xwayland shouldn't be enabled. If no package.use.force is used, the REQUIRED_USE would be more sane

---

This is also due to my preference, on top of the technical necessity with package.use.force approach.
However the USE flag description should mention this somehow - that the USE flag is only meaningful with USE=wayland. Perhaps something like an "if wayland is enabled" in the end, but currently that would be repetitive with the rest. Maybe something like

> <flag name="xwayland">Enable X11 application support for Wayland sessions</flag>

is good enough, as I'm not sure how to write a good English sentence with the conditional that isn't so repetitive. Or maybe something like

> <flag name="xwayland">Enable X11 application support for <use>wayland</use> sessions</flag>

---

The second variant leads to
```
PkgInvalidXml: package metadata.xml violates metadata.xsd:
line 12, col 0: (SCHEMAV_ELEMENT_CONTENT) Element 'use': This element is not expected. Expected is one of ( cat, pkg ).
```
going with the first for now